### ISSUE TITLE
Cycleway overlay: cycle track with walking allowed

### ISSUE THREAD

**Issue by westnordost** [streetcomplete/StreetComplete#5991]
### Discussed in https://github.com/streetcomplete/StreetComplete/discussions/5990

<div type='discussions-op-text'>

<sup>Originally posted by **olecky** October 30, 2024</sup>
When editing in the cycleways overlay, there doesn't seem to be an option to select cycle track (signed with [D4](https://commons.wikimedia.org/wiki/File:Vienna_Convention_road_sign_D4a.svg) or similar sign), respecting the fact that pedestrians are allowed to use it because of national or international legislation. In most cases, if there is no sidewalk, pedestrians are allowed to walk on the cycle track (article 20 paragraph 3 of the Convention on Road Traffic). Example:

![IMG_20200927_145527-resized](https://github.com/user-attachments/assets/2ae7e379-837b-4f15-b57c-4711ae1ccb50)

In the app, the closest I can choose are:

- EXCLUSIVE - which adds foot=no, banning pedestrians
- NON_SEGREGATED - which adds foot=designated, implying that there is sign designating the highway for pedestrians (so a combination of D4 and D5 rather than only D4)

Both of the option seem incorrect to me in the example shown on the picture. I'm not sure if it's necessary to add one more option to distinguish cycle tracks with walking allowed and not. From my point of view, removing the foot=no addition in EXCLUSIVE would solve the problem. Is it really useful for pedestrian routing in situations where there is parallel cycle track and footpath?

</div>

**Comment by westnordost** [comment]
So, I think @olecky's suggestion makes sense - remove `foot` when retagging to `highway=cycleway`. 

@olecky: Care to implement this yourself and submit a PR, as you already looked at the source code?

**Comment by olecky** [comment]
I did the edit I suggested, but I didn't test it, so feel free to reject the pull request.

**Comment by westnordost** [comment]
Yeah, it doesn't work, see my comment in #5992.

Maybe it needs to be discussed whether the new behavior should be 
a) always remove `foot`
b) remove `foot` if it is not `no`

**Comment by olecky** [comment]
I'm a bit afraid of removing foot, because occasionally there are also cases with:

a) signs explicitly forbidding pedestrians from the track (for example: https://www.openstreetmap.org/way/826279591) 
b) panels under cycle track sign explicitly mentioning that pedestrians are allowed (similar to https://wiki.openstreetmap.org/wiki/File:Z239ZZ1022_10GehwegRadfahrerFrei.jpg, but the other way around)

**Comment by wielandb** [comment]
> In most cases, if there is no sidewalk, pedestrians are allowed to walk on the cycle track (article 20 paragraph 3 of the Convention on Road Traffic)

Are we sure this applys to germany (or other countries) as well? If not, I think removing foot=no from the way (even if the other tags should still be enough to infer the rules) wouldn't be a good idea.

**Comment by matkoniecz** [comment]
it applies in Poland

if reference "article 20 paragraph 3 of the Convention on Road Traffic" is accurate then it is likely to be shared quite widely.

**Comment by olecky** [comment]
It applies also in Belgium: [Art. 9. – Plaats op de openbare weg](https://www.wegcode.be/nl/regelgeving/2024005817~0mocswfbry#ybwltapvzm)

and Netherlands: [Artikel 4 Reglement verkeersregels en verkeerstekens 1990 (RVV 1990)](https://wetten.overheid.nl/jci1.3:c:BWBR0004825&hoofdstuk=II&paragraaf=1&artikel=4&z=2024-07-01&g=2024-07-01)

**Comment by mnalis** [comment]
**TL;DR**: probably it would need extra option (to distinguish "exclusive to only cyclist" vs. "cyclist but pedestrians allowed"), as there is a use case where there is explicit sign prohibiting pedestrians, in which case pedestrians can not legally use that cycleway even when there is no suitable sidewalk.

---

For reference,  [article 20 paragraph 3 of the Convention on Road Traffic](https://treaties.un.org/doc/Treaties/1977/05/19770524%2000-13%20AM/Ch_XI_B_19.pdf) says:

> 3. If it is not possible to use pavements (side-walks) or verges, or if none is provided, pedestrians may walk on the carriageway; where there is a cycle track and the density of traffic so permits, they may walk on the cycle track, but shall not obstruct cycle and moped traffic in doing so.

However I'm not sure that is encodified fully in Croatian law. There is exception for using the **road** when there is no sidewalk in Članak 125. [ZoSPnC](https://www.zakon.hr/z/78/Zakon-o-sigurnosti-prometa-na-cestama) ((translation)):

>  (1) As an exception to Article 124, paragraphs 1 and 3 of this Act, on a road where there is no sidewalk or other surface designated, or suitable for pedestrian movement, or on a road where there is a sidewalk or other surface designated, or suitable for pedestrian movement that pedestrians cannot use for any reason, pedestrians can move on the roadway.

However, there seems to be no such exception for cycleways (bicycle tracks), and while walking on bicycle lanes (being part of the roadway) seems allowed in such circumstances, using bicycle tracks (i.e. cycleways _separated_ from roadway) in fact seems explicitly prohibited by  Članak 129. later, which only allows crossing, and not walking on cycleway:

> (1) Pedestrians are obliged to cross the roadway and bicycle track or lane carefully and by the shortest route, after making sure that they can do so safely before entering the roadway.

But I'm not a lawyer... 

 (translation note:  _"suitable"_ is not perfect translation for _"prikladan"_, which is perhaps better translated as _"appropriate"_ or _"corresponds to requirement"_)

**Comment by olecky** [comment]
> **TL;DR**: probably it would need extra option (to distinguish _"exclusive to only cyclist"_ vs. _"for cyclist but pedestrians allowed"_), as there is another use case where there is explicit sign prohibiting pedestrians

And why simply to not change the foot tag (leave it as it is) in case the cycle track option is selected?

There might also be other signs adjusting access rights (allowing or disallowing for example speed pedelecs, different moped categories, vehicles of the residents, etc.) I don't think it makes sense to create a separate option for each possible combination. 

If it's an important question for pedestrian routing, then maybe we need a separate quest "Are pedestrians allowed to walk on the cycle track here?"

**Comment by mnalis** [comment]
> probably it would need extra option (to distinguish "exclusive to only cyclist" vs. "for cyclist but pedestrians allowed"), as there is another use case where there is explicit sign prohibiting pedestrians, in which case pedestrians may not legally use that cycleway 

On the other hand, such situation is probably less common, so we can solve the conundrum by  just never adding `foot=no` (and remove _"exclusive"_ wording from answer) in all cases, and let user leave a note and/or some other mapper[^1] add extra restrictions if they want to indicate that detail.

It should be ***significantly less taxing to SC mappers*** that way, and even in such cases (of explicit sign prohibiting pedestrians) would **not** tag ***incorrectly*** (but just **sub-optimally**)


[^1]:   e.g. [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) allows adding  many extra access tags to ways, including `foot=no`, via `Uh...` answer `Add access...`)

**Comment by matkoniecz** [comment]
Looking at my Todo/idea list I had request to distinguish "designated for pedestrians and cyclists", "cycleway, pedestrians allowed" (bicycle icon) and "cycleway, pedestrians banned" (bicycle icon + crossed over foot icon) 

The last is rare but happens

**Comment by wielandb** [comment]
So, throwing my two cents in here:

In germany, using a cycleway that is marked as an exclusive cycleway is indeed forbidden for pedestrians (German Traffic Code, Appendix 2, Sign 237). (I was astonished to learn through this discussion that that seems to be not the norm and that the Vienna convention suggests something else) 

I think we should keep in mind the considerations we had to take when I edited the cycleway overlay recently.
We came to the conclusion that we cannot expect the average SC user to know the law (and know it correctly enough) to let them tag anything law-related in SC.
So we always only tag what is _signed_ on the way. E.g. changing a cycleway to a footway will remove the tags indicating it is signed as such, but `bicycle=yes` will not be removed. (Which actually might not be that good now that I think about it...)

I am actually okay with removing foot=no, because such things should be in the data consumers hand. The data consumer should know the default access restrictions for the country anyway. 

Anyway, for @matkoniecz, if you do that, maybe also go down the "signed"-route. :)

**Comment by matkoniecz** [comment]
"cycleway, but a sign allows pedestrians"

the problem is that in many places blue sign with bicycle only allows pedestrians if there is no footway along

see https://github.com/streetcomplete/StreetComplete/issues/5991#issue-2624320142 (if it would be in Poland)

> In germany, using a cycleway that is marked as an exclusive cycleway is indeed forbidden for pedestrians

also without footway along it?

**Comment by wielandb** [comment]
> also without footway along it?

As it seems, yes

**Comment by wielandb** [comment]
> the problem is that in many places blue sign with bicycle only allows pedestrians if there is no footway along
I think here we come back to the problem of SC users not having to know the legal situation. The SC interface should only ask for something that everyone can assess. That's why the bicycle overlay always states "there is a sign...".

E.g., in Australia as I have learned, cycling on the sidewalk is always allowed (unless explicitly forbidden). But for example I as a foreigner would not know that, and if it was just about "Is it allowed or not?", I would probably tag "no" (which would be wrong). By always asking for a sign, I would select "Its a path without a sign about bicycles", and the data consumer has to know if according to legislation in that country, bicycles can use that path or not. No matter if I personally know the law or not.

**Comment by westnordost** [comment]
I agree with @wielandb. That leaves us in a bit of a pickle if we don't want to add yet two more options:
- bike path, no signs
- bike path, sign prohibits pedestrians
- bike path, sign allows pedestrians
And, yeah, we don't want that.

So, I think the best compromise would be:
1. rename option from "bike-only path" to "bike path"
2. remove `foot` only if it was `designated` before, otherwise leave it alone
    **OR alternatively**
    remove `foot` always except if `foot:signed=yes`

What do you think is the better compromise?

**Comment by westnordost** [comment]
Both will be problematic for another reason, though: `highway=cycleway + foot=yes` will be parsed as `Designated shared-use path`... which is... not wrong, when just looking at the tags, I'd say.

**Comment by matkoniecz** [comment]
> I think here we come back to the problem of SC users not having to know the legal situation. The SC interface should only ask for something that everyone can assess. That's why the bicycle overlay always states "there is a sign...".

what about having dedicated behaviour depending on what signs mean in a given country?

**Comment by westnordost** [comment]
Bad idea. In countries where it is generally allowed, there can be a pedestrian forbidden sign and the other way round.

**Comment by matkoniecz** [comment]
I meant that there would be different options to select. Or that "bicycles only" would have different icons depending on location.

**Comment by mnalis** [comment]
**TL;DR:** I'd prefer adding extra options to allow fully specifying situation in cycleway overlay. 

> I think here we come back to the problem of SC users not having to know the legal situation. The SC interface should only ask for something that everyone can assess. That's why the bicycle overlay always states "there is a sign...".

Absolutely agree.

> I meant that there would be different options to select. Or that "bicycles only" would have different icons depending on location.

Country specific config would have to allow three options (`foot_on_cycleway_forbidden_by_default`, `foor_on_cycleway_allowed_by_default` and `default_differs_by_region` with last one being the default for all countries that do not have specific information; in which case we should display all three options). It is too much work for too little benefit, IMHO.

> I agree with @wielandb. That leaves us in a bit of a pickle if we don't want to add yet two more options:
> * bike path, no signs
> * bike path, sign prohibits pedestrians
> * bike path, sign allows pedestrians
>
> And, yeah, we don't want that.

As an avid cyclist as well as and advanced OSM (and SCEE) user with heavy focus on bicycle infrastructure, I'm obviously an outlier. That being said, I see basically three general approaches:

- have user **specify exact situation on the ground regarding pedestrian usage of cycleways** (i.e. offer all three options above), even if that makes cycleway overlay more complex
- completely **ignore pedestrian usage on cycleways** always (i.e. in cycleway- ignore and never touch `foot` tags, change all wordings to be bicycle only and explicitly indicate the pedestrians are irrelevant; e.g. _"Bicycle path, regardless if pedestrians are also allowed or not"_ )
- ask for only some bicycle+foot combinations, and ignore others (which might exist), asking users to leave notes in such cases (or give up in frustration on that overlay). I.e. keep status quo.

---

Now, I (being biased) obviously prefer the first option as it allows to specify exact situation, even if it introduces two new options. But I think more novice users of cycleway might too prefer two more options in exchange for having clear-cut choice without current confusion (especially in countries that allow walking on cycleways unless explicitly forbidden)

**Comment by westnordost** [comment]
Well, this is the bicycle overlay, not the bicycle and foot overlay. I don't want to litter this with even more selection options, it's already _unübersichtlich_¹ as it is.
From the point of view of a bicyclist, it matters little whether pedestrians are also _allowed_ on a bike path as long as the path hasn't been actually designated for pedestrians, too. After all, bike paths in countries where pedestrians are allowed by default and bike paths in countries where they aren't are not really... different.

Plus, for bike paths that are also designated to pedestrians, i.e. are equals with the bicycle traffic, we already do have a selection option (shared-use bike and footpath).

Anyway, I tend towards:
1. rename option from "bike-only path" to "bike path"
2. remove foot only if it was `designated` before, otherwise leave it as before
3. interpret `bicycle=designated + foot=yes + segregated ≠ yes` as "bike path" rather than "shared-use path". 

---

¹ actually there is no good translation to this in English. Best described as "difficult to get an overview over"

**Comment by mnalis** [comment]
**TL;DR:** name "cycleway overlay" is misnomer, it is already equally (and probably more) about pedestrians. So IMHO it should represent cycleway equivalents for already existing pedestrian answers too. Perhaps split pedestrian-related answer and bicycle-related answer (with UI like in surface overlay for segregated bicycle+foot path)? 

> Well, this is the bicycle overlay, not the bicycle and foot overlay

Well, perhaps it should be renamed then? 

I mean, out of the 8 offered answers, 5 explicitly mention it being **designated** for **pedestrians** , and pedestrian access is implied in 2 more answers, with only 1 answer rejecting pedestrian access (and that one - `Bike-only path` - is currently being suggested to be repurposed to **_also_** sometimes mean to be allowed for pedestrians). 

For comparison, only 4 answers are about being designated for bicycles, so it is actually more of pedestrian-overlay than cycleway-overlay already.

> Plus, for bike paths that are also designated to pedestrians, i.e. are equals with the bicycle traffic, we already do have a selection option (shared-use bike and footpath).

... and those are **_exactly the options_** which are causing the confusion; as it conflicting with claim above that this overlay is _only_ for bicycles. 

Because, having an option for _"Designated shared-use path"_ (and others) by definition involves pedestrians, and that creates confusion: yeah, it is _called_ "bicycle overlay", but it _also_ offers various degrees of shared bike&pedestrian paths, so it is about both, but is my situation _"path for bicycles but pedestrians allowed"_ more like _"shared bicycle & pedestrian path"_ or more like _"bicycle path"_? It is a very hard question (especially for not-so-proficient mapper).

As for the existing answers:
- we have _"Footway, no explicit sign about cycling (whether it is allowed depends on legislation)"_, but not _"Cycleway, no explicit sign about pedestrians  (whether it is allowed depends on legislation)"_ ?
- we have _"Footway, but a sign allows cycling"_ , but not _"Cycleway, but a sign allows pedestrians"_ ?
- we have _"(footway and) a sign explicitly prohibits cycling"_, but not _"cycleway and a sign explicitly prohibits pedestrians"_
 
Especially if it is called (as we're being reminded) **cycleway** overlay? Maybe we should have little more cycleway representation in there, then? :smiley: 

- one way to avoid confusion is to give more options which would make it perfectly clear which one to choose as suggested above. But yeah, I get your point that it gets crowded there. But there are many possible options :man_shrugging:  
  Perhaps it could be split in sub-answers (yeah I know guidelines, but we do that sometimes - e.g. plastic in recycling quests when plastic is chosen asks sub-answers about which types of plastic are allowed). 
  So have just 3 initial answers in first sub-answer: `bicycles are explicitly designated by sign here`, `bicycles are allowed by sign here, but it is not designated for them`, `bicycles are explicitly forbidden by sign here`, `bicycles use is not signed here, depends on legislation`  followed by similar (but extended to include segregated=yes/no option) sub-answers about pedestrians.
  Or,  **even nicer, it could be in the form of UI like the one we have for surface overlay for segregated bicycle+foot paths**: have left column for bicycle answers, and right column for pedestrian answer. Like this (texts and pictures are just examples):
![small_bicycle_pedestrian_sc](https://github.com/user-attachments/assets/3777700f-2e53-421b-be7c-fba7f4d15f38)


<details>

<summary> other, inferior options, mentioned for completeness, collapsed to make the post seem smaller</summary>

- another way to be clear might be to ***really*** make it about **bicycles only**, i.e. **never** look nor update any tag related to pedestrians, and get rid of "shared path" answer and make info / other options clear that "nobody cares about what pedestrians can or cannot do in _this_ overlay". That would of course imply a need for other footway overlay (or repurposing of sidewalk overlay for pedestrian-questions, then).

- third way, i.e. if it continues to be about _"mostly about bicycles, but sometimes also pedestrians, in fact, mostly about pedestrians and sometimes bicycles"_ , I'm afraid that the confusion will continue as _"when should pedestrians be included and when not"_ would likely continue to be unclear. We might perhaps somewhat improve it, but probably not clear the core confusion.

</details>

> rename option from "bike-only path" to "bike path"
> remove foot only if it was designated before, otherwise leave it as before

I get the idea, but seems too "automagical" to me, i.e. have unintended/unknown (by mapper) consequences.
And, it would not solve the core confusion anyway IMHO.

>    interpret bicycle=designated + foot=yes + segregated ≠ yes as "bike path" rather than "shared-use path".

In any case, I'd only consider it as "shared-use path" if it is `bicycle=designated` + `foot=designated`. If it is only `bicycle=designated` (and `foot` missing or anything other then `designated`) then it is _"bike path"_ .

**Comment by westnordost** [comment]
Only because pedestrians are mentioned in the answers does not mean it is *about* pedestrians. The point of view is that of cyclists, and that reflects in the possible answer options (e.g. "not made for cyclists (could be allowed or disallowed)").

While a foot+bike overlay that merges the sidewalk overlay and the bike overlay may be worth some consideration, it is a lot of effort to design a new overlay and I am not willing to throw the towel on meaningfully separating these two concerns (pedestrians, cyclists) in two separate overlays. I will certainly not trash this whole overlay because of this ticket, as it is well possible to handle it, in my opinion.
mnalis, you are free to brainstorm for a new and improved bike+pedestrian overlay, of course, but then first best in the forum or the discussion board. A note on your mock up: Open question is how to specify whether these are segregated, and what to specify when there are no signs (e.g. "just a path", "just a (foot?)way").

**Comment by westnordost** [comment]
>>  interpret `bicycle=designated` + `foot=yes` + `segregated ≠ yes` as "bike path" rather than "shared-use path".

> In any case, I'd only consider it as "shared-use path" if it is `bicycle=designated` + `foot=designated`. If it is only `bicycle=designated` (and foot missing or anything other then designated) then it is "bike path" .

Hmm, this makes 
```
highway = path
bicycle = designated
segregated = no
```
be interpreted as "bike path". (`=path` is interpreted as `foot=yes`, `segregated=no` is not `≠ yes`)
Hmm, I guess, that's alright, though. That's what the tags say, after all. To interpret `highway = path` as `foot=somewhere_inbetween_yes_and_designated` would be weird.

**Comment by wielandb** [comment]
> Only because pedestrians are mentioned in the answers does not mean it is _about_ pedestrians. The point of view is that of cyclists, and that reflects in the possible answer options (e.g. "not made for cyclists (could be allowed or disallowed)").
> 
> While a foot+bike overlay that merges the sidewalk overlay and the bike overlay may be worth some consideration, it is a lot of effort to design a new overlay and I am not willing to throw the towel on meaningfully separating these two concerns (pedestrians, cyclists) in two separate overlays. I will certainly not trash this whole overlay because of this ticket, as it is well possible to handle it, in my opinion. mnalis, you are free to brainstorm for a new and improved bike+pedestrian overlay, of course, but then first best in the forum or the discussion board. A note on your mock up: Open question is how to specify whether these are segregated, and what to specify when there are no signs (e.g. "just a path", "just a (foot?)way"). Also, note that in the bike overlay, we have two different forms: one for bike infrastructure mapped on the road way, one for bike infrastructure mapped on separate paths.

So, it seems that I have not received the notifications that the discussion had resumed, and thus did not see that @mnalis had already made a mock-up of a new version of the overlay (and that the issue was already closed) and so I made a little mock-up of what I thought a new "paths overlay" could look like. It may address some of the concerns, so I am leaving it here for completeness' sake. Maybe @mnalis can use it as further inspiration for further discussions. 

![paths_overlay_draft](https://github.com/user-attachments/assets/d448ff0f-428d-4519-94ac-f4f7cc62fad0)
